docs/adding_a_task.md

```diff
@@ -0,0 +1,252 @@
+
+# Adding a Task
+
+To add a new task to MTEB, you need to do three things:
+
+1) Implement a task with the desired dataset, by subclassing an abstract task
+2) Add metadata to the task
+3) Submit the edits to the [MTEB](https://github.com/embeddings-benchmark/mteb) repository
+
+If you have any questions regarding this process feel free to open a discussion [thread](https://github.com/embeddings-benchmark/mteb/discussions). 
+
+> Note: When we mention adding a dataset we refer to a subclass of one of the abstasks.
+
+## 1) Creating a new subclass
+
+### A Simple Example
+
+To add a new task, you need to implement a new class that inherits from the `AbsTask` associated with the task type (e.g. `AbsTaskReranking` for reranking tasks). You can find the supported task types in [here](https://github.com/embeddings-benchmark/mteb-draft/tree/main/mteb/abstasks).
+
+```python
+from mteb import MTEB
+from mteb.abstasks.AbsTaskReranking import AbsTaskReranking
+from sentence_transformers import SentenceTransformer
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+class SciDocsReranking(AbsTaskReranking):
+    metadata = TaskMetadata(
+        name="SciDocsRR",
+        description="Ranking of related scientific papers based on their title.",
+        reference="https://allenai.org/data/scidocs",
+        hf_hub_name="mteb/scidocs-reranking",
+        type="Reranking",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["en"],
+        main_score="map",
+        revision="d3c5e1fc0b855ab6097bf1cda04dd73947d7caab",
+        date=None,
+        form="written",
+        domains=["Academic", "Non-fiction"],
+        task_subtypes=["Scientific Reranking"],
+        license="cc-by-4.0",
+        socioeconomic_status="high",
+        annotations_creators=None,
+        dialect=None,
+        text_creation="found",
+        bibtex_citation= ... # removed for brevity
+)
+
```

Comment:
Maybe this file should be called `adding_a_dataset` as the above is a new dataset I think?

Then `adding_a_task` would be about e.g. adding Retrieval or Classification etc.

Reply:
👍; Since the mmteb readme distinguishes `Contribute a new task` & `Contribute a new dataset`, would go with `adding_a_dataset` for this file and maybe we have another file `adding_a_task` in case someone wants to add a new embedding task (I'm sure there are others besides CLF, Clustering etc.)

Reply:
I will rename it to adding_a_dataset. I think most users comfortable enough to add a new abstask, won't need too much help so don't think we need to add an introduction there.